docs/user-guide/a1-01-matrix-layout.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: user-guide
+---
+
+Handling Matrix Layout Differences on Different Platforms
+============================
+
+The differences on default matrix layout or storage conventions between GLSL (OpenGL/Vulkan) and HLSL has been an issue that frequently causes confusion among developers. When writing applications that work on different targets, one important goal that developers frequently seek is to make it possible to pass the same matrix generated by host code to the same shader code, regardless of what graphics API is being used (e.g. Vulkan, OpenGL or Direct3D). As a solution to shader cross-compilation, Slang provides necessary tools for developers navigate around the differences between GLSL and HLSL targets.
+
+Row-major vs Column-major storage layouts
+-------------------------
+In this article, the term matrix layouts or storage layouts refers to the order of matrix elements in memory. Specifically, we use the following umambiguous technique to determine whether a matrix is in `row-major` and `column-major` layout: if we have a 4x4 matrix defined as `float m[16]`, and the translation terms are stored at `m[12], m[13], m[14]`, then we say this matrix is in `column-major` layout. If the translation terms are stored at `m[3], m[7], m[11]`, then we say this matrix is in `row-major` layout.
```

Comment:
This definition of row and column major seems to be the opposite of what I'd expect? 

https://en.wikipedia.org/wiki/Row-_and_column-major_order

It is consistent with the later code in this doc. 

Though if I was using 'row major' on HLSL, I'd expect to be doing 

```
float4 result = mul(vertexPosition, modelViewProjectionMatrix);
```

Perhaps all the flipping comes out in the wash, in that the code above in Slang, will be flipped in the HLSL output (to give what you have below), and then the flipping of the definition of row major makes that all alright...